services/supervisor/tests/supervisor/migrations/mod.rs

```diff
@@ -0,0 +1,879 @@
+
+    // Request migration to 0.5.
+    // This migration will require two migration requests.
+    let deadline_height = DEADLINE_HEIGHT;
+    let request = MigrationRequest {
+        new_artifact: MigrationServiceV05.artifact_id(),
+        service: MigrationService::INSTANCE_NAME.into(),
+        deadline_height,
+    };
+
+    let api = testkit.api();
+    let tx_hash = request_migration(&api, request.clone());
+    let block = testkit.create_block();
+
+    block[tx_hash]
+        .status()
+        .expect("Transaction should be executed successfully");
+
+    // After the first migration step, version should be "0.2".
+    wait_for_migration_success(
+        &mut testkit,
+        deadline_height,
+        request,
+        Version::new(0, 2, 0),
+    );
+
+    let snapshot = testkit.snapshot();
+    let prefixed = Prefixed::new(MigrationService::INSTANCE_NAME, snapshot.as_ref());
+
+    migration_service::v02::verify_schema(prefixed);
+
+    // Request the same migration.
+    let deadline_height = Height(DEADLINE_HEIGHT.0 * 2);
+    let request = MigrationRequest {
+        new_artifact: MigrationServiceV05.artifact_id(),
+        service: MigrationService::INSTANCE_NAME.into(),
+        deadline_height,
+    };
+
+    let api = testkit.api();
+    let tx_hash = request_migration(&api, request.clone());
+    let block = testkit.create_block();
+
+    block[tx_hash]
+        .status()
+        .expect("Transaction should be executed successfully");
+
+    // Now we finally should have version "0.5".
+    wait_for_migration_success(
+        &mut testkit,
+        deadline_height,
+        request,
+        Version::new(0, 5, 0),
+    );
+
+    let snapshot = testkit.snapshot();
+    let prefixed = Prefixed::new(MigrationService::INSTANCE_NAME, snapshot.as_ref());
+
+    migration_service::v05::verify_schema(prefixed);
+}
+
+/// This test checks that attempt to request a migration for service that doesn't support
+/// migrations results in a migration failure.
+#[test]
+fn no_migration_support() {
+    let mut testkit = testkit_with_supervisor_and_service_no_migrations(1);
+
+    // Stop service instance before running the migration.
+    stop_service(&mut testkit, MigrationService::INSTANCE_ID);
+
+    // Request migration.
+    let deadline_height = DEADLINE_HEIGHT;
+    let request = MigrationRequest {
+        new_artifact: MigrationServiceV05.artifact_id(),
+        service: MigrationService::INSTANCE_NAME.into(),
+        deadline_height,
+    };
+
+    let api = testkit.api();
+    let tx_hash = request_migration(&api, request.clone());
+    let block = testkit.create_block();
+
+    // Despite the fact that migration should fail, the transaction with request
+    // should be executed successfully.
+    block[tx_hash]
+        .status()
+        .expect("Transaction should be executed successfully");
+
+    let error = wait_for_migration_fail(&mut testkit, deadline_height, request);
```

Comment:
Shouldn't the migration marked as failed right away? I think it could be beneficial to check that it is indeed so.